File: orchalanguage-spring-integration-implementation/src/main/java/orcha/lang/compiler/referenceimpl/springIntegration/SendInstructionFactory.java
package orcha.lang.compiler.referenceimpl.springIntegration;

import org.springframework.beans.factory.config.AbstractFactoryBean;

public class SendInstructionFactory extends AbstractFactoryBean<SendInstructionForSpringIntegration> {

    public SendInstructionFactory(){
        setSingleton(false);
    }

    @Override
    public Class<?> getObjectType() {
        return SendInstructionForSpringIntegration.class;
    }

    @Override
    protected SendInstructionForSpringIntegration createInstance() throws Exception {
        return new SendInstructionForSpringIntegration();
    }
}

File: orchalang-kafka-implementation/src/main/kotlin/orcha/lang/compiler/kafka/SendInstructionFactory.kt
package orcha.lang.compiler.kafka

import org.springframework.beans.factory.config.AbstractFactoryBean

class SendInstructionFactory : AbstractFactoryBean<SendInstructionForSpringIntegration>() {
    init {
        isSingleton = false
    }

    override fun getObjectType(): Class<*>? {
        return SendInstructionForSpringIntegration::class.java
    }

    @Throws(Exception::class)
    override fun createInstance(): SendInstructionForSpringIntegration {
        return SendInstructionForSpringIntegration()
    }
}
